src/v/cloud_roles/refresh_credentials.cc

```diff
@@ -241,6 +242,40 @@ http::client cloud_roles::refresh_credentials::impl::make_api_client() const {
       _as};
 }
 
+cloud_roles::refresh_credentials cloud_roles::make_refresh_credentials(
+  model::cloud_credentials_source cloud_credentials_source,
+  ss::gate& gate,
+  ss::abort_source& as,
+  credentials_update_cb_t creds_update_cb,
+  s3::aws_region_name region,
+  std::optional<endpoint> endpoint,
+  retry_params retry_params) {
+    switch (cloud_credentials_source) {
+    case model::cloud_credentials_source::config_file:
+        vlog(
+          clrl_log.error,
+          "invalid request to create refresh_credentials for static "
+          "credentials");
+        throw std::invalid_argument(
+          "cannot generate refresh with static credentials");
```

Comment:
you can use fmt_with_ctx to format strings with extra context. that plus here you might want some extra info to help in debugging?

Reply:
thanks for the tip, fixed